cranelift/codegen/src/machinst/helpers.rs

```diff
@@ -0,0 +1,35 @@
+//! Miscellaneous helpers for machine backends.
+
+use crate::ir::types::*;
+use crate::ir::Type;
+
+/// Returns the size (in bits) of a given type.
+pub fn ty_bits(ty: Type) -> usize {
+    match ty {
+        B1 => 1,
+        B8 | I8 => 8,
+        B16 | I16 => 16,
+        B32 | I32 | F32 | R32 => 32,
+        B64 | I64 | F64 | R64 => 64,
+        B128 | I128 => 128,
+        IFLAGS | FFLAGS => 32,
+        B8X8 | I8X8 | B16X4 | I16X4 | B32X2 | I32X2 => 64,
+        B8X16 | I8X16 | B16X8 | I16X8 | B32X4 | I32X4 | B64X2 | I64X2 => 128,
+        F32X4 | F64X2 => 128,
+        _ => panic!("ty_bits() on unknown type: {:?}", ty),
+    }
+}
+
+pub(crate) fn ty_is_int(ty: Type) -> bool {
+    match ty {
+        B1 | B8 | I8 | B16 | I16 | B32 | I32 | B64 | I64 | R32 | R64 => true,
+        F32 | F64 | B128 | I128 | I8X8 | I8X16 | I16X4 | I16X8 | I32X2 | I32X4 | I64X2 | F32X4
+        | F64X2 => false,
+        IFLAGS | FFLAGS => panic!("Unexpected flags type"),
+        _ => panic!("ty_is_int() on unknown type: {:?}", ty),
+    }
```

Comment:
```suggestion
    !ty.is_vector && (ty.is_bool() || ty.is_int() || ty.is_ref())
```

Reply:
Personally I prefer the original, since it involves less indirection for the reader and doesn't make any assumptions about `is_vector`, `is_bool`, `is_int` and `is_ref` covering all possible types exactly once.

Reply:
Hmm, I think I'm going to go with @bjorn3's suggestion here, just because we've had several instances of merge conflicts in these match-on-type statements when someone adds support for a new vector type. It seems to me that it's better to have a single source of truth for the type categories, which is what we have with `Type::is_*()`.